server/controllers/profileController.js

```diff
@@ -0,0 +1,70 @@
+const Profile = require("../models/ProfileModel");
+const mongoose = require("mongoose");
+
+const toObjectId = mongoose.Types.ObjectId;
+
+// create new profile using given arguments
+
+exports.createProfile = async (req, res, next) => {
+  try {
+    const { firstName, lasttName, phone, address, gender, description } =
+      req.body;
+    const data = {
+      userId: toObjectId(req.params.id) || "",
```

Comment:
This field is crucial because it helps us determine which profile belongs to which user. We wouldn't want to store an empty string if this field has not been specified.